### PR TITLE
Fixing redudant notices when installing plugins via payment task list

### PR DESCRIPTION
**Pull request by joelclimbsthings**
So I got sidetracked today by the fact that this has been annoying me:

image

This is the case if just one plugin resides within the `plugins` array for a given gateway, which seemed a bit off. Turns out it was just resulting from some rapid unnecessary rerenders/remounts, which I've tried to address with some memoization.

And now:
image

🥳 

### Detailed test instructions:

-   Activate remote tester plugin
-   Checkout this branch
- Set country to South Africa for testing gateway support
- Go to task list -> payments -> payfast
- Ensure only one notice prints after the plugin is installed

### CODE REVIEW

File: client/task-list/tasks/payments/RemotePayments/components/PaymentMethod.js [hunk 1]
```diff
@@ -92,8 +92,8 @@ export const PaymentMethod = ( {
 		.map( ( pluginSlug ) => pluginNames[ pluginSlug ] )
 		.join( ' ' + __( 'and', 'woocommerce-admin' ) + ' ' );
 
-	const installStep =
-		plugins && plugins.length
+	const installStep = useMemo( () => {
```
Comment (joshuatf) [inline client/task-list/tasks/payments/RemotePayments/components/PaymentMethod.js:95]: Curious why we should `useMemo` around the whole step versus using `React.memo` for just the `Plugin` component.  Is this a re-render issue with the plugin component or something else?

**Review by joshuatf — COMMENTED** [review]
Thanks for fixing this, Joel!  💯 

This is testing well, just left a question about how this works and one possible alternative.

File: client/task-list/tasks/payments/RemotePayments/components/PaymentMethod.js (hunk 1 above)
Comment (joelclimbsthings) [reply client/task-list/tasks/payments/RemotePayments/components/PaymentMethod.js:95]: There actually was excessive rendering down the entire chain due to redeclaring functions on every render, forcing every component that accepts those functions to rerender as well. 

I actually experimented with `React.memo`, but it didn't solve the issue. I just tried again as you described but I'm still seeing _way_ too many remounts for the `<Stepper />` component, which ends up remounting `<Plugins />` in turn:

image

That's with this code: 
```js
const MemoizedPlugins = memo( Plugins );

const installStep =
	plugins && plugins.length
		? {
				key: 'install',
				label: sprintf(
					/* translators: %s = one or more plugin names joined by "and" */
					__( 'Install %s', 'woocommerce-admin' ),
					pluginNamesString
				),
				content: (
					<MemoizedPlugins
						onComplete={ ( installedPlugins, response ) => {
							createNoticesFromResponse( response );
							recordEvent(
								'tasklist_payment_install_method',
								{
									plugins,
								}
							);
						} }
						onError={ ( errors, response ) =>
							createNoticesFromResponse( response )
						}
						autoInstall
						pluginSlugs={ plugins }
					/>
				),
				isComplete: ! pluginsToInstall.length,
			}
		: null;
```

I don't have a lot of experience with these memoization tools, but my suspicion was just that the `installStep` becomes a new object on every render, which in turn causes excessive remounts down the line. Let me know if I'm doing anything amiss above, but memoizing the `installStep` object and the `<DefaultStepper />` based on that seemed to do the trick.

**Comment by joelclimbsthings** [comment]
Much thanks @joshuatf for the review! I've responded to your comment. Let me know what you think.

**Review by joshuatf — APPROVED** [review]
That makes sense.  Thanks for the explanation, Joel!  Let's go ahead and get this merged then as it does seem to fix the issue.  🚢

**Pull request merged**